src/main/scala/com/gu/mobile.notifications.client/ApiClient.scala

```
@@ -1,33 +1,46 @@
 package com.gu.mobile.notifications.client
 
+import com.gu.mobile.notifications.client.models.legacy.Notification
+
 import scala.concurrent.{ExecutionContext, Future}
-import dispatch._
 import play.api.libs.json._
-import models.JsonImplicits._
-import lib.const
 import models._
 
-trait ApiClient {
-  implicit val executionContext: ExecutionContext
+trait HttpProvider {
+
+  sealed trait HttpResponse
+  /** For status >= 200 and < 300 */
+  case class HttpOk(status: Int, body: String) extends HttpResponse
+  /** For every other status */
+  case class HttpError(status: Int, body: String) extends Throwable with HttpResponse
+  
+  case class ContentType(mediaType: String, charset: String)
+
+  def post(url: String, contentType: ContentType, body: Array[Byte]): Future[HttpResponse]
+  def get(url: String): Future[HttpResponse]
+}
 
+trait ApiClient extends HttpProvider {
   /** Host of the Guardian Notifications Service */
   def host: String
 
-  /** Http client */
-  def httpClient: Http
-
-  def healthcheck: Future[Healthcheck] = Http((url(host) / "healthcheck") OK as.String) recover {
-    case StatusCode(code) => Unhealthy(code)
-  } map const(Ok)
+  def healthcheck(implicit ec: ExecutionContext): Future[Healthcheck] = {
+    get(s"$host/healthcheck").map {
+      case HttpOk(200, body) => Ok
+      case HttpOk(code, _) => Unhealthy(Some(code))
+      case httpError => Unhealthy()
```

Why not return the error code if it is a HttpError?